src/newlines.cpp

```diff
@@ -3231,7 +3231,7 @@ void newlines_chunk_pos(c_token_t chunk_type, tokenpos_e mode)
    chunk_t *prev;
    int     nl_flag;
 
-   if ((mode & (TP_JOIN | TP_LEAD | TP_TRAIL)) == 0)
+   if ((mode & (TP_JOIN | TP_LEAD | TP_TRAIL)) == 0 && chunk_type != CT_COMMA)
```

Comment:
Well, I was just saying that we should have coverage on the original bad behaviour (i.e. to show there was a bug and we had to fix it). If that's not possible with the other option it's ok.